src/main/java/io/stargate/sgv2/jsonapi/service/operation/embeddings/FindEmbeddingProvidersOperation.java

```diff
@@ -67,10 +69,16 @@ private record EmbeddingProviderResponse(
       List<EmbeddingProvidersConfig.EmbeddingProviderConfig.ParameterConfig> parameters,
       List<ModelConfigResponse> models) {
     private static EmbeddingProviderResponse provider(
-        EmbeddingProvidersConfig.EmbeddingProviderConfig embeddingProviderConfig) {
+        EmbeddingProvidersConfig.EmbeddingProviderConfig embeddingProviderConfig,
+        boolean returnDeprecatedModels) {
       ArrayList<ModelConfigResponse> modelsRemoveProperties = new ArrayList<>();
       for (EmbeddingProvidersConfig.EmbeddingProviderConfig.ModelConfig model :
           embeddingProviderConfig.models()) {
+        if (!returnDeprecatedModels) {
+          if (model.deprecated().isPresent() && model.deprecated().get()) {
```

Comment:
idea: I think this can be shortened to `model.deprecated().orElse(false)`

Reply:
Yes, changed it [b122f24](https://github.com/stargate/data-api/pull/1641/commits/b122f2486eb18931877194414fd4694b0714fcdf)